### PR TITLE
New dialogs for the profile

### LINKED ISSUE #5663: Update to newer confirmation dialogs on Profile page
## The request

While most of Home Assistant has been updated to use the newer / nicer confirmation dialog prompts introduced by Timmo in https://github.com/home-assistant/frontend/pull/4053 , the profile page hasn't yet received that same treatment. They're still using the older and uglier browser based alert prompts. These should get updated so everything is consistent.

**Examples**

Deleting refresh tokens:

image

Deleting long lived access tokens:

image

Creating long lived access tokens:

image

The "Log Out" button on this page uses the new style, the above ones should be updated to match.

image

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Move dialogs to new style

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: closes #5663
- This PR is related to issue:
- Link to documentation pull request:

For the push notifications, I made another PR earlier <https://github.com/home-assistant/frontend/pull/5778>

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/profile/ha-long-lived-access-tokens-card.js [hunk 1]
```diff
@@ -102,22 +106,25 @@ class HaLongLivedTokens extends LocalizeMixin(EventsMixin(PolymerElement)) {
...
     if (!name) return;
     try {
       const token = await this.hass.callWS({
         type: "auth/long_lived_access_token",
         lifespan: 3650,
         client_name: name,
       });
-      prompt(
-        this.localize(
+      await showPromptDialog(this, {
```
Comment (bramkragten) [inline src/panels/profile/ha-long-lived-access-tokens-card.js:121]: Should this be a prompt? The old alerts couldn't be copied, the new ones can I think?

File: src/panels/profile/ha-refresh-tokens-card.js [hunk 2]
```diff
@@ -91,27 +95,30 @@ class HaRefreshTokens extends LocalizeMixin(EventsMixin(PolymerElement)) {
...
         type: "auth/delete_refresh_token",
-        refresh_token_id: ev.model.item.id,
+        refresh_token_id: token.id,
       });
       this.fire("hass-refresh-tokens");
     } catch (err) {
       // eslint-disable-next-line
       console.error(err);
-      alert(this.localize("ui.panel.profile.refresh_tokens.delete_failed"));
+      await showAlertDialog(this, {
```
Comment (bramkragten) [inline src/panels/profile/ha-refresh-tokens-card.js:119]: No need to await
```suggestion
     showAlertDialog(this, {
```

File: src/panels/profile/ha-long-lived-access-tokens-card.js (hunk 1 above)
Comment (ludeeus) [reply src/panels/profile/ha-long-lived-access-tokens-card.js:121]: It can, but should it?
image

Comment (bramkragten) [reply src/panels/profile/ha-long-lived-access-tokens-card.js:121]: That looks bad 😄

**Review by bramkragten — APPROVED** [review]

File: src/panels/profile/ha-long-lived-access-tokens-card.js (hunk 1 above)
Comment (ludeeus) [reply src/panels/profile/ha-long-lived-access-tokens-card.js:121]: Way cleaner with prompt
image

**Pull request merged**